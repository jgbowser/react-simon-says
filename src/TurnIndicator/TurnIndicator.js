import React from 'react'

export default class TurnIndicator extends React.Component {
  
  render() {

    const message = this.props.player === 'COMPUTER' ? "Simon's turn. Pay attention!" : "Your turn. Good luck!" 
    return (
      <div className="turn_indicator">
        <p>{message}</p>
      </div>
    )
  }
}