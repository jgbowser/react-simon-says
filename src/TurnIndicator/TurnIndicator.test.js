import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import TurnIndicator from './TurnIndicator'

describe('TurnIndicator Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<TurnIndicator />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  it('renders Your turn. Good Luck! when passed PLAYER as a prop', () => {
    const tree = renderer
      .create(<TurnIndicator player="PLAYER" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders Simons turn... when passed COMPUTER as a prop', () => {
    const tree = renderer
      .create(<TurnIndicator player="COMPUTER" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})