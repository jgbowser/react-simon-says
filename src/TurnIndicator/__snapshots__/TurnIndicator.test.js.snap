// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TurnIndicator Component renders Simons turn... when passed COMPUTER as a prop 1`] = `
<div
  className="turn_indicator"
>
  <p>
    Simon's turn. Pay attention!
  </p>
</div>
`;

exports[`TurnIndicator Component renders Your turn. Good Luck! when passed PLAYER as a prop 1`] = `
<div
  className="turn_indicator"
>
  <p>
    Your turn. Good luck!
  </p>
</div>
`;
